LabExT/Movement/MoverNew.py

```diff
@@ -602,17 +615,40 @@ def lift_stages(self):
         Performs NO safe movement (i.e no Path planning)
         """
 
-        def _lift_lower_stages(lift):
+        def _toggle_stage_lift():
+            """
+            Moves all stages to z lift level.
+            Moves up if stages_are_lifted is false and vice versa.
+            """
             for calibration in self.calibrations.values():
+                _is_lifted = calibration in self._lifted_stages
+                target_z = self.z_lift if not _is_lifted else 0
                 with calibration.perform_in_chip_coordinates():
-                    calibration.move_absolute(
-                        calibration.get_position() + ChipCoordinate(z=lift))
-
-        _lift_lower_stages(self.z_lift)
+                    stage_start_pos = calibration.get_position()
+                    target_coordinate = ChipCoordinate(
+                        x=stage_start_pos.x, y=stage_start_pos.y, z=target_z)
+                    try:
+                        calibration.move_absolute(
+                            target_coordinate, wait_for_stopping=True)
+                    finally:
+                        stage_end_pos = calibration.get_position()
+                        lift_delta = stage_end_pos - stage_start_pos
+
+                        if not np.isclose(
```

Comment:
@marcoep @boris-vukovic Need same feedback on the tolerances.
```
absolute(a - b) <= (atol + rtol * absolute(b))
```
`atol` and `rtol` are the default ones at the moment.